finality-aleph/src/session.rs

```diff
@@ -29,25 +27,44 @@ impl<B: Block> SessionBoundaries<B> {
     }
 }
 
-pub fn first_block_of_session<N: AtLeast32BitUnsigned>(
-    session_id: SessionId,
-    period: SessionPeriod,
-) -> N {
-    (session_id.0 * period.0).into()
+fn first_block_of_session(session_id: SessionId, period: SessionPeriod) -> BlockNumber {
+    session_id.0 * period.0
 }
 
-pub fn last_block_of_session<N: AtLeast32BitUnsigned>(
-    session_id: SessionId,
-    period: SessionPeriod,
-) -> N {
-    ((session_id.0 + 1) * period.0 - 1).into()
+fn last_block_of_session(session_id: SessionId, period: SessionPeriod) -> BlockNumber {
+    (session_id.0 + 1) * period.0 - 1
 }
 
-pub fn session_id_from_block_num<N: AtLeast32BitUnsigned>(
-    num: N,
-    period: SessionPeriod,
-) -> SessionId {
-    SessionId((num / period.0.into()).saturated_into())
+fn session_id_from_block_num(num: BlockNumber, period: SessionPeriod) -> SessionId {
+    SessionId(num / period.0)
+}
+
+pub struct SessionInfo {
+    session_period: SessionPeriod,
+}
+
+impl SessionInfo {
+    pub fn new(session_period: SessionPeriod) -> Self {
+        Self { session_period }
+    }
+}
+
+/// Struct for getting the session boundaries.
+impl SessionInfo {
+    /// Returns session id of the session that block belongs to.
+    pub fn session_id_from_block_num(&self, n: BlockNumber) -> SessionId {
+        session_id_from_block_num(n, self.session_period)
```

Comment:
Eh, I'm pretty sure this would be better defined just internally, without a separate function. It seems the only reason to have the separate function (although not even for this specific case!) is `SessionBoundaries`, but that I would say should be returned by a method of `SessionInfo`, e.g. `boundaries_for_session(&self, id: SessionId) -> SessionBoundaries`. You don't need the separate functions then (I think?).

Reply:
+1

Reply:
Great idea, will do

Reply:
Sure, sounds good. I only want it done, doesn't have to be in this PR. :P